polyhost/device/hid_helper.py

```
@@ -119,6 +120,25 @@ def get_console_output(self):
 
+    def peek_notification(self) -> None:
+        """Non-blocking read; sets _firmware_ready_pending if FIRMWARE_READY is seen."""
+        if self.interface is None:
+            return
+        try:
+            with self.lock:
+                report = self.interface.read(self.settings.HID_REPORT_SIZE, timeout=0)
+            if report and report.startswith(b"P\x17"):
+                self._firmware_ready_pending = True
+        except Exception:
+            pass
```

Ruff (0.15.12)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)